File: Two_Pointers/088.Merge-Sorted-Array/088.Merge-Sorted-Array_v2.ts
function merge(nums1: number[], m: number, nums2: number[], n: number): void {
    let p1 = m - 1;
    let p2 = n - 1;
    for (let i = nums1.length - 1; i >= 0; i--) {
        if (p1 < 0) {
            nums1[i] = nums2[p2--];
        } else if (p2 < 0) {
            nums1[i] = nums1[p1--];
        } else if (nums1[p1] > nums2[p2]) {
            nums1[i] = nums1[p1--];
        } else {
            nums1[i] = nums2[p2--];
        }
    }
};

File: Two_Pointers/088.Merge-Sorted-Array/088.Merge-Sorted-Array_v2.java
class Solution {
    public void merge(int[] nums1, int m, int[] nums2, int n) {
        int i = m - 1;
        int j = n - 1;

        for (int k = nums1.length - 1; k >= 0; k--) {
            if (i < 0) {
                nums1[k] = nums2[j];
                j -= 1;
            } else if (j < 0) {
                nums1[k] = nums1[i];
                i -= 1;
            } else if (nums1[i] >= nums2[j]) {
                nums1[k] = nums1[i];
                i -= 1;
            } else {
                nums1[k] = nums2[j];
                j -= 1;
            }
        }
    }
}